xmtp/src/client.rs

```diff
@@ -21,9 +29,21 @@ where
     pub network: Network,
     pub persistence: NamespacedPersistence<P>,
     pub account: VmacAccount,
+    pub(super) _store: S,
```

Comment:
>  that delegates more logic around how the data is structured to the store itself rather than the client.

Exactly. This follows a more rustic approach,  similar to the trait Into.

> I guess the advantage over defining a Storage trait that has methods like storeNewDecryptedMessage, storeBlah, is that the Client can blindly call .store() on any object it has?

Correct. All details are abstracted into the impl.

> Is there a name for this pattern?

https://rust-unofficial.github.io/patterns/patterns/behavioural/strategy.html

Reply:
>why is it pub(super) as opposed to just private?
Currently the builder requires access to it as we are directly constructing instances of Client.